inductiva/tasks/task.py

```diff
@@ -172,13 +172,64 @@ def wait(self, polling_period: int = 5) -> models.TaskStatusCode:
 
             time.sleep(polling_period)
 
-    def kill(self) -> None:
-        """Kill the task.
-
-        This method issues a request to the API to kill the task. It doesn't
-        block waiting for confirmation if the task was killed.
+    def kill(self,
+             wait: bool = False,
+             timeout: float = 5,
+             maxretries: int = 5) -> bool:
+        """Tries to kill a task.
+        If wait is False (default value) the function sends the
+        kill command and returns. If wait is True the function
+        checks for confirmation that the kill command was successfull.
+        Args:
+            wait: Determines if we wait for the kill command to
+                finish.
+            timeout: Time in seconds to wait for the kill command.
+            maxretries: Number of times to retry the kill command.
+        Returns:
+            True if the kill command was successfull, False otherwise.
         """
-        self._api.kill_task(path_params=self._get_path_params())
+        while maxretries > 0:
```

Comment:
We should check if the task is not in a terminal status. If it is, the endpoint to kill will bug, and print several messages like:

```bash
Something went wrong while sending the kill command.Retrying...
Something went wrong while sending the kill command.Retrying...
Something went wrong while sending the kill command.Retrying...
Something went wrong while sending the kill command.Retrying...
Something went wrong while sending the killcommand. Max retries reached.
False
```

When it should print, `Task has already terminated with status %s.`

Reply:
Done